core/controlplane/config/config.go

```diff
@@ -435,6 +438,7 @@ type DeploymentSettings struct {
 	CloudWatchLogging       `yaml:"cloudWatchLogging,omitempty"`
 	AmazonSsmAgent          `yaml:"amazonSsmAgent,omitempty"`
 	CloudFormationStreaming bool `yaml:"cloudFormationStreaming,omitempty"`
+	KubeDnsMasq             `yaml:"kubeDnsMasq,omitempty"`
```

Comment:
My thinking was that we should try to place new switches into top level groups , so that we can add more related options should we need to do so.

```
kubeDns:  # kube-dns related opts follow
  nodeLocalResolver: true    # run dnsmasq locally on every node
  monitorDomainsResoveTime:  # domains to be picked up by Prometheus
     -  ec2.amazonaws.com
     -  mycompany.internal.domain
     -  google.com
```